provider/mesh/peer.go

```diff
@@ -0,0 +1,241 @@
+package mesh
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/prometheus/alertmanager/provider"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/log"
+	"github.com/prometheus/common/model"
+	"github.com/satori/go.uuid"
+	"github.com/weaveworks/mesh"
+)
+
+type NotificationInfos struct {
```

Comment:
`NotificationsInfo` would be better, as this is information about multiple notifications rather than many pieces of information about one notification.